sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -2422,6 +2422,9 @@ class ReadFromBigQuery(PTransform):
       to run queries with INTERACTIVE priority. This option is ignored when
       reading from a table rather than a query. To learn more about query
       priority, see: https://cloud.google.com/bigquery/docs/running-queries
+    output_type (str): By default, the schema returned from this transform
+      would be of type PYTHON_DICT. Other schema types can be specified
+      ("BEAM_ROW").
```

Comment:
@pabloem this PR is about ready. I just wanted to check - are you OK with this change to the API?

Reply:
Hmm yeah I suppose this makes sense. It would be nice if the Beam rows would also support the Python dictionary interface but I suppose we've already considered this to be too difficult? LGTM

Reply:
It actually should be doable, we're tracking schema inference for TypedDict in https://github.com/apache/beam/issues/22086. Once that's done, we could produce a TypedDict typehint for PYTHON_DICT, and both modes would support schema inference.

Unfortunately I think #22086 a ways out still, so it makes sense to have this as a stopgap.

Reply:
we don't need to have schema inference for this, though, right? We just need to ensure the output values look like dicts to consumer transforms, look like namedtuples to the pcollection coder, and the schema coder should be set for the pcollection - is that it? may not need to wait for #22086?

However, like I said, I don't really know all of the implications of actually implementing those things - so if it's too much trouble, I'm okay with moving forward as-is. Thanks!

Reply:
Thanks I appreciate the ideas here :)

It's not as simple as just setting the coder, RowCoder is parameterized by the schema: https://github.com/apache/beam/blob/master/sdks/python/apache_beam/coders/row_coder.py#L52-L59

https://github.com/apache/beam/blob/d51b497fb229d75eef8b7baee98cdb817a592a58/model/pipeline/src/main/proto/org/apache/beam/model/pipeline/v1/beam_runner_api.proto#L1059

So we need to be able to infer the schema at pipeline construction time in order to build one.